internal/performance/performance.go

```diff
@@ -145,14 +153,26 @@ func (c *PerfCollector) processEntityMetrics(metricsValues *types.PerfEntityMetr
 			c.logger.Debugf("The metric: %v is not containing at least one sample, this is not expected", name)
 			continue
 		}
-		// MaxSamples is set to 1 but the API is retreiving multiple samples with the same value for historical interval metrics.
+
+		// MaxSamples is set to 1 but the API is retrieving multiple samples with the same value for historical interval metrics.
 		// We will take just first one.
 		metricVal = metricValueSeries.Value[0]
+
+		// This is a short-lived object, the purpose is to compute the average of the different performance metrics
+		// when more than one instance per entity returns a value
+		if _, ok := accumulateMetrics[name]; !ok {
+			accumulateMetrics[name] = &Accumulator{}
+		}
+		accumulateMetrics[name].Occurrences++
+		accumulateMetrics[name].Sum += metricVal
+	}
+	for key, val := range accumulateMetrics {
 		perfMetricsByRef[metricsValues.Entity] = append(perfMetricsByRef[metricsValues.Entity], PerfMetric{
-			Counter: name,
-			Value:   metricVal,
+			Counter: key,
+			Value:   val.Sum / val.Occurrences,
```

Comment:
could you add a unit test to validate this logic? if not too complex

Reply:
yes sure, I will add some use cases. I wanted to validate first the approach

Reply:
Added @gsanchezgavier 
Tell me if you think they are enough